website/client/src/components/auth/socialAuth/google.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div
+    :class="{
+      'btn': !isHomePage,
+      'btn-secondary': !isHomePage,
+      'social-button': true,
+      'social-button-home': isHomePage,
+    }"
+    @click="socialAuth('google')"
+  >
+    <div
+      class="svg-icon social-icon"
+      v-html="icons.googleIcon"
+    ></div>
+    <span>{{ (registering || isHomePage || isGroupPlansPage)
+      ? $t('signUpWithSocial', {social: 'Google'})
+      : $t('loginWithSocial', {social: 'Google'}) }}</span>
+  </div>
+</template>
+
+<style lang="scss">
+  .social-button {
+    width: 100%;
+    height: 100%;
+    white-space: inherit;
+    text-align: center;
+
+    .text {
+      display: inline-block;
+    }
+  }
+
+  .social-button-home {
+    border-radius: 2px;
+    border: solid 2px #bda8ff;
+    width: 100%;
+    min-height: 40px;
+    padding: .5em;
+    background: transparent;
+    margin-bottom: .5em;
+    color: #bda8ff;
+    transition: .5s;
+
+    span {
+      transition: none;
+    }
+  }
+
+  .social-icon {
+    margin-left: 1em;
+    margin-right: 1em;
+    width: 18px;
+    height: 18px;
+    display: inline-block;
+    vertical-align: top;
+    margin-top: .1em;
+  }
+</style>
+
+<script>
+import hello from 'hellojs';
+import { setUpAxios } from '@/libs/auth';
+import googleIcon from '@/assets/svg/google.svg';
+
+export default {
+  data () {
+    const data = {};
+    data.icons = Object.freeze({
+      googleIcon,
+    });
+    return data;
+  },
+  computed: {
+    registering () {
+      if (this.$route.path.startsWith('/register')) {
+        return true;
+      }
+      return false;
+    },
+    isHomePage () {
+      if (this.$route.path.startsWith('/static/home')) {
+        return true;
+      }
+      return false;
+    },
+    isGroupPlansPage () {
+      if (this.$route.path.startsWith('/static/plan')
+      || this.$route.path.startsWith('/static/group-plans')) {
+        return true;
+      }
+      return false;
+    },
+  },
+  mounted () {
+    hello.init({
+      facebook: process.env.FACEBOOK_KEY, // eslint-disable-line
+      // windows: WINDOWS_CLIENT_ID,
+      google: process.env.GOOGLE_CLIENT_ID, // eslint-disable-line
+    });
+  },
+  methods: {
+    async socialAuth (network) {
```

Comment:
I think this part of the social auth process is shared between facebook and google entirely, partially for apple, could it be abstracted for example using a mixin?

Reply:
yep, one thing of the remain works is to abstract the validate logic and some auth logic, i will finish later this week.